src/commons.rs

```diff
@@ -554,3 +556,66 @@ pub extern "C" fn z_str_null() -> z_owned_str_t {
 pub extern "C" fn z_str_loan(s: &z_owned_str_t) -> *const libc::c_char {
     s._cstr
 }
+
+#[repr(C)]
+#[derive(Clone, Copy, Debug)]
+pub enum zc_locality_t {
+    ANY = 0,
+    SESSION_LOCAL = 1,
+    REMOTE = 2,
+}
+
+impl From<Locality> for zc_locality_t {
+    fn from(k: Locality) -> Self {
+        match k {
+            Locality::Any => zc_locality_t::ANY,
+            Locality::SessionLocal => zc_locality_t::SESSION_LOCAL,
+            Locality::Remote => zc_locality_t::REMOTE,
+        }
+    }
+}
+
+impl From<zc_locality_t> for Locality {
+    fn from(k: zc_locality_t) -> Self {
+        match k {
+            zc_locality_t::ANY => Locality::Any,
+            zc_locality_t::SESSION_LOCAL => Locality::SessionLocal,
+            zc_locality_t::REMOTE => Locality::Remote,
+        }
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn zc_locality_default() -> zc_locality_t {
+    Locality::default().into()
+}
+
+#[repr(C)]
+#[derive(Clone, Copy, Debug)]
+pub enum zc_reply_keyexpr_t {
```

Comment:
Similar to `Locality`, also `ReplyKeyExpr` is marked unstable for the time being.